crates/miden-testing/src/kernel_tests/tx/test_account.rs

```diff
@@ -890,10 +887,36 @@ fn test_get_storage_commitment() -> anyhow::Result<()> {
             assert_eqw.err="actual storage commitment is not equal to the expected one"
         end
         "#,
-        expected_storage_commitment = &account.storage().commitment(),
+        expected_storage_commitment = &tx_context.account().storage().commitment(),
     );
     tx_context.execute_code(&code)?;
 
+    Ok(())
+}
+
+/// This test creates an account with mock storage slots and calls the
+/// `compute_storage_commitment` procedure each time the storage is updated.
+///
+/// Namely, we invoke the `mock_account::compute_storage_commitment` procedure:
+/// - Right after the account creation.
+/// - After updating the 0'th storage slot.
+/// - Right after the previous call to make sure it returns the same commitment from the cached
+///   data.
+/// - After updating the 1'st storage slot.
+#[test]
+fn test_compute_storage_commitment() -> anyhow::Result<()> {
+    let tx_context = TransactionContextBuilder::with_existing_mock_account().build().unwrap();
+    let mut account_clone = tx_context.account().clone();
+    let account_storage = account_clone.storage_mut();
+
+    let init_storage_commitment = account_storage.commitment();
+
+    account_storage.set_item(0, [9, 10, 11, 12].map(Felt::new).into())?;
+    let storage_commitment_0 = account_storage.commitment();
+
+    account_storage.set_item(1, [13, 14, 15, 16].map(Felt::new).into())?;
+    let storage_commitment_1 = account_storage.commitment();
```

Comment:
Let's change one of these to `set_map_item` (slot index 2), so we make sure that the commitment is correctly updated both for `set_item` and `set_map_item`.